src/charm.py

```diff
@@ -1459,15 +1445,33 @@ def _was_restore_successful(self, container: Container, service: ServiceInfo) ->
             logger.debug("Restore check early exit: Patroni has not started yet")
             return False
 
+        restoring_backup = self.app_peer_data.get("restoring-backup")
+        restore_timeline = self.app_peer_data.get("restore-timeline")
+        restore_to_time = self.app_peer_data.get("restore-to-time")
+        # Sometimes this check fails to connect to the PostgreSQL. Not happened on VM, so it's a k8s-specific try-catch
+        # try:
+        #     current_timeline = self.postgresql.get_current_timeline()
+        # except PostgreSQLGetCurrentTimelineError:
+        #     logger.debug("Restore check early exit: can't get current wal timeline")
```

Comment:
Did you manage to figure out and fix in this PR the reason why it was failing on K8s? If so, instead of commenting, you should remove this block of code.

Reply:
I do not have this issue while testing manually, but in the integration tests this error occurred several times. I think it caused by #718 and safe to remove.